pandas/tseries/period.py

```diff
@@ -946,6 +946,27 @@ def repeat(self, n, *args, **kwargs):
         # overwrites method from DatetimeIndexOpsMixin
         return self._shallow_copy(self.values.repeat(n))
 
+    def where(self, cond, other=None):
+        """
+        .. versionadded:: 0.18.2
+
+        Return an Index of same shape as self and whose corresponding
+        entries are from self where cond is True and otherwise are from
+        other.
```

Comment:
@sinhrks I wish that could get some more uniformity w.r.t. Periods. I realize that there is a bug here with ``other`` which I fixed in ``tseries/base.py/.where``, end up duplicating code here.